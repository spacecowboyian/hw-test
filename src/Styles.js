import { css } from 'emotion/macro';

export const colorCyan = '#00FFDE';

export const colorSeries = [
  "#003D40",
  "#006667",
  "#008F91",
  "#00B0B2",
  "#00C7B1",
  "#00E9CB",
  "#2FA200",
  "#00CE5F",
  "#00E834",
  "#00FF39"
];

export const hiddenAboveMobile = css`
  @media(min-width: 600px) {
    display: none;
  }
`

export const spacing = '4px';

export const standardBox = css`
  padding:calc(${spacing}*4) calc(${spacing}*8);
`
