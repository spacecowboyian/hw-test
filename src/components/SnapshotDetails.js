import React from 'react'
import { Row, Col } from 'react-flexbox-grid';
import { css } from 'emotion/macro';
import { Doughnut } from 'react-chartjs-2';

import { spacing, colorSeries } from '../Styles';

const snapshotTitle = css`
  font-size: 30px;
  font-weight: 100;
`

const keyList = css`
  margin:0;
`

const keyItem = css`
  padding-bottom: calc(${spacing}*3)
`;

const keyDotBase = css`
  display: inline-block;
  width: 8px;
  height: 8px;
  overflow: hidden;
  border-radius: 8px;
  background-color: black;
`

const keyDotContent = css`
  display: block;
  margin-top:100px;
`

const keyDot = color => css`
  ${keyDotBase}
  background-color: ${colorSeries[color]}
`

const keyDescription = css`
  display: inline-block;
  margin-left: calc(${spacing}*3);
`

const chartColumn = css`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  color: #515A5A;
`

const chartContainer = css`
  position:relative;
`

const chartPercentage = css`
 position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  font-size:40px;
  font-weight: 100;
`

const chartTotal = css`
  font-size: 24px;
  font-weight: 600;
  margin-top: calc(${spacing}*8);
`

const chartTotalLabel = css`
  font-weight: 600;
  margin-top: calc(${spacing}*2);
`

const chartOptions = {
  cutoutPercentage: 80,
}

export default function SnapshotDetails({ data }) {
  return (
    <React.Fragment>
      <h3 className={snapshotTitle}>Premium Range</h3>
      <Row>
        <Col xs={6}>
          <dl className={keyList}>
            <div className={keyItem}>
              <dt className={keyDot(0)}>
                <span className={keyDotContent}>Range 1</span>
              </dt>
              <dd className={keyDescription}>
                $0 - $5000
                        </dd>
            </div>
            <div className={keyItem}>
              <dt className={keyDot(2)}>
                <span className={keyDotContent}>Range 2</span>
              </dt>
              <dd className={keyDescription}>
                $5000 - $10,000
                        </dd>
            </div>

          </dl>
        </Col>
        <Col className={chartColumn} xs={6}>
          <div className={chartContainer}>
            <Doughnut data={data} legend={null} height={170} options={chartOptions} />
            <div className={chartPercentage}>95%</div>
          </div>
          <div className={chartTotal}>$1,548,990</div>
          <div className={chartTotalLabel}>Total Premiums</div>
        </Col>
      </Row>
    </React.Fragment>
  )
}
