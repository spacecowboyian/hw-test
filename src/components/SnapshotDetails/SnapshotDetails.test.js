import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { colorSeries } from '../../Styles';
import SnapshotDetails from './SnapshotDetails';

afterEach(cleanup);

const chartData = {
  labels: [
    'Red',
    'Green',
    'Yellow'
  ],
  datasets: [{
    data: [300, 50, 100, 60, 85, 200],
    backgroundColor: colorSeries,
    borderWidth: 0
  },]
};

it('renders correctly', () => {
  const tree = renderer
    .create(<SnapshotDetails data={chartData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
