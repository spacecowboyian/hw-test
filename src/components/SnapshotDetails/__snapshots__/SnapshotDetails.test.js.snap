// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h3
    className="css-svz9iy-snapshotTitle"
  >
    Premium Range
  </h3>,
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <dl
        className="css-1024hqm-keyList"
      >
        <div
          className="css-qw6ik-keyItem"
        >
          <dt
            className="css-1uar067-keyDotBase"
          >
            <span
              className="css-18wopn0-keyDotContent"
            >
              Range 1
            </span>
          </dt>
          <dd
            className="css-73s3vi-keyDescription"
          >
            $0 - $5000
          </dd>
        </div>
        <div
          className="css-qw6ik-keyItem"
        >
          <dt
            className="css-1mmkf41-keyDotBase"
          >
            <span
              className="css-18wopn0-keyDotContent"
            >
              Range 2
            </span>
          </dt>
          <dd
            className="css-73s3vi-keyDescription"
          >
            $5000 - $10,000
          </dd>
        </div>
      </dl>
    </div>
    <div
      className="col-xs-6 css-1nk341g-chartColumn"
    >
      <div
        className="css-1symqm2-chartContainer"
      >
        <div
          className="css-uy1i6m-chartPercentage"
        >
          95%
        </div>
      </div>
      <div
        className="css-16mqj27-chartTotal"
      >
        $1,548,990
      </div>
      <div
        className="css-1xay42j-chartTotalLabel"
      >
        Total Premiums
      </div>
    </div>
  </div>,
]
`;
